### ISSUE TITLE
CKEditor mac safari screen hang when I click on editor section any text

### ISSUE THREAD

**Issue by RanaVikash** [ckeditor/ckeditor-dev#552]
## Are you reporting a feature or a bug?

Bug. 

## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

*Put all reference links here…*

## Provide detailed reproduction steps (if any)

1.Add any html as source 
2.When you check it in editor the html loads
3. Click on any text and mac safari browser screen hangs (Freeze

[hanging html.txt](https://github.com/ckeditor/ckeditor-dev/files/1094574/hanging.html.txt)

**Comment by wojtekw92** [comment]
confirmed on Safari Version 10.1.1 (12603.2.4)

**Comment by gitgauravhub** [comment]
Hi wojtekw92
Is this issue going to be resolved in near feature. I am facing it on same version of Safari you mentioned,
This is hanging up the whole screen leaving the whole web page non responsive with a Spinning pinwheel coming up for infinite times. 
Please let us know.
Thanks

**Comment by artemsk** [comment]
It seems this is the same issue as in #471

**Comment by mlewand** [comment]
Thanks @artemsk, could very well be a duplicate of ticket you have mentioned. @wojtekw92 please check whether it's a duplicate as I don't have a Safari atm (just check if it was reproducible with 4.6.2, if not it's much likely a `tableselection` regression).

**Comment by wojtekw92** [comment]
@mlewand, I checked this on http://cdn.ckeditor.com/4.6.2/full-all/samples/ with safari 10.1.1 and the browser is hanging I think that this issue is duplicate of #471.

**Comment by mlewand** [comment]
Thanks for the confirmation, so let's track this issue in #552.

**Comment by andrewboni** [comment]
@mlewand also running into this on both `4.5.5` as well as `4.8.0` in Safari 11

You can also [reproduce it here](https://ckeditor.com/latest/samples/old/replacebycode.html) with this markup:

```html
<table>
	<tbody>
		<tr>
			<td>
			<table>
				<tbody>
					<tr>
						<td>
						<table>
							<tbody>
							</tbody>
						</table>
						</td>
					</tr>
				</tbody>
			</table>
			</td>
		</tr>
	</tbody>
</table>
```

Any thoughts on a resolution?

**Comment by mlewand** [comment]
It should be resolved by #471. I don't have any assumptions here, only after a proper investigation we would be able to tell something. Currently we're focused on 4.9.0 tickets.

**Comment by andrewboni** [comment]
@mlewand this still seems to be an issue even with the latest version. Steps to reproduce:

1. Go to https://sdk.ckeditor.com/samples/classic.html
2. Paste the following code in Source:
```html
<table>
	<tbody>
		<tr>
			<td>
			<table>
				<tbody>
					<tr>
						<td>
						<table>
							<tbody>
							</tbody>
						</table>
						</td>
					</tr>
				</tbody>
			</table>
			</td>
		</tr>
	</tbody>
</table>
```
3. Click Source again (to return to WYSIWYG mode)
4. Safari crashes (I'm on v11.0.2, but I believe this affects many / all versions)

Let me know if you'd prefer me to open up a new ticket.

**Comment by mlewand** [comment]
@andrewboni this issue is tracked in #471. This safari issue is still reproducible as #471 is not resolved. Please subscribe mentioned issue instead of this to get any updates on that matter.
